packages/transaction_signing_gateway/lib/transaction_broadcaster.dart

```diff
@@ -1,11 +1,11 @@
+import 'package:alan/alan.dart';
 import 'package:dartz/dartz.dart';
 import 'package:transaction_signing_gateway/model/private_account_credentials.dart';
 import 'package:transaction_signing_gateway/model/signed_transaction.dart';
 import 'package:transaction_signing_gateway/model/transaction_broadcasting_failure.dart';
-import 'package:transaction_signing_gateway/model/transaction_hash.dart';
 
 abstract class TransactionBroadcaster {
-  Future<Either<TransactionBroadcastingFailure, TransactionHash>> broadcast({
+  Future<Either<TransactionBroadcastingFailure, TxResponse>> broadcast({
```

Comment:
Why are we needing to return a class of the type from inside the generated files? Is there a problem in returning a type of our own class?
Also, this is affecting the abstraction as well. Technically, the abstraction shouldn't return a very specific type (and that too from the generated files).
I don't know the context of the bug, but do we have a strong reason of doing this?
cc: @andrzejchm

Reply:
I just read the bug description, we should allow the previous class to have all the properties of the full transaction object, rather than returning the object of a generated file. Technically refactoring `TransactionHash` to `TransactionResponse` of `Transaction` simply and keeping all the parameters required there.

Reply:
sure we can do that

Reply:
I agree with @wal33d006 we should wrap the Transaction into our own class with all the relevant fields copied over. Like:

```dart
class TransactionResponse extends Equatable {
  final TransactionHash hash;
  final String data;
  final List<TransactionLog> logs;
  final Decimal gasUsed;
  final Decimal gasWanted;
  final DateTime timestamp;
}
```

Reply:
done this and will soon push it

Reply:
I created a factor constructor that takes in `TxResponse` and spits out  `TransactionResponse` 

```dart
 factory TransactionResponse.fromTxResponse(TxResponse tx) {
    return TransactionResponse(
      hash: TransactionHash(hash: tx.txhash),
      data: tx.data,
      rawLog: tx.rawLog,
      logs: List<TransactionLog>.from(
        tx.logs.map(
          (log) => TransactionLog(
            log: log.log,
            msgIndex: log.msgIndex,
            events: List.from(
              log.events.map(
                (e) => e.toProto3Json(),
              ),
            ),
          ),
        ),
      ),
      gasUsed: Decimal.parse('${tx.gasUsed}'),
      gasWanted: Decimal.parse('${tx.gasWanted}'),
      timestamp: DateTime.fromMillisecondsSinceEpoch(int.parse(tx.timestamp)),
    );
  }


```

Reply:
Hi @wal33d006 waiting for your review